// [START declaration]
syntax = "proto3";
package sapient_msg;

import "google/protobuf/descriptor.proto";
// [END declaration]

// [START java_declaration]
option java_multiple_files = true;
option java_package = "uk.gov.dstl.sapientmsg";
option java_outer_classname = "ProtoOptionsProto";
// [END java_declaration]

// [START messages]
message FileOptions {
    string standard_version = 1;
}

message ValidationOptions  {
    optional bool is_mandatory = 1;
    optional bool is_ulid = 2;
    optional bool is_uuid = 3;
    optional string xml_name = 4;
    optional bool xml_is_attribute = 5;
    optional bool xml_ignore = 6;
    optional string xml_singly_nested = 7;
    optional bool xml_is_parent_value = 8;
    optional string enum_name = 9;
    optional bool is_proto_time = 10;
    optional bool xml_tentative = 11;
    optional bool is_time = 12;
}

message MessageOptions {
  optional string xml_message_name = 50001;
  optional bool is_sapient_message = 50002;
}

extend google.protobuf.FileOptions {
    FileOptions file_options = 100001;
}

extend google.protobuf.EnumValueOptions {
    optional ValidationOptions enum_options = 50001;
}

extend google.protobuf.OneofOptions {
    optional ValidationOptions oneof_options = 50001;
}

extend google.protobuf.FieldOptions {
    optional ValidationOptions field_options = 50001;
}

extend google.protobuf.MessageOptions {
  optional MessageOptions message_options = 50001;
}
// [END messages]
